src/store/storage-controller.ts

```diff
@@ -84,12 +85,11 @@ export class StorageController {
 
     const messages = await messageStore.query(tenant, filter);
 
+    // for every message, only include the data as `encodedData` if the data size is equal or smaller than the size threshold
     for (const message of messages) {
       const dataCid = message.descriptor.dataCid;
-      if (dataCid !== undefined) {
-        // TODO: #219 (https://github.com/TBD54566975/dwn-sdk-js/issues/219)
-        // temporary placeholder for keeping status-quo of returning data in `encodedData`
-        // once #219 is implemented, `encodedData` may or may not exist directly as part of the returned message here
+      const dataSize = message.descriptor.dataSize;
```

Comment:
why do we assume `dataCid` might be undefined but we assert that `dataSize` is defined? Does `dataCid` being defined guarantee that `dataSize` is also defined? If so, where does that check happen?

Reply:
> Does `dataCid` being defined guarantee that `dataSize` is also defined?

Yes.

> If so, where does that check

https://github.com/TBD54566975/dwn-sdk-js/blob/main/json-schemas/records/records-write.json#L74